chrome/content/zotero/zoteroPane.js

```diff
@@ -4514,35 +4520,39 @@ var ZoteroPane = new function()
 	 * @returns {Promise<Zotero.Item[] | null>}
 	 */
 	this.addAttachmentFromDialog = async function (link, parentItemID, files = null) {
+		var libraryID;
+		if (Zotero_Tabs.selectedType === 'library') {
+			let collectionTreeRow = this.getCollectionTreeRow();
+			if (link) {
+				if (collectionTreeRow.isWithinGroup()) {
+					Zotero.alert(null, "", "Linked files cannot be added to group libraries.");
+					return null;
+				}
+				else if (collectionTreeRow.isPublications()) {
+					Zotero.alert(
+						null,
+						Zotero.getString('general.error'),
+						Zotero.getString('publications.error.linkedFilesCannotBeAdded')
+					);
+					return null;
+				}
+			}
```

Comment:
This is here because there's no way through the UI to add a linked file from a reader tab?

It makes sense to check `isPublications()` here, since that by definition requires a collection tree row, but we should probably move the linked-file check down and use `Zotero.Libraries.get(libraryID).isGroup` instead of `isWithinGroup()`, in case some plugin triggers this function from the reader or it gets exposed in some other way.